src/components/interface/cbuf_mgr/cos_alloc.c

```diff
@@ -249,9 +250,14 @@ static void _alloc_libc_free(void *ptr)
 		__alloc_t *pointer = BLOCK_START(ptr);
 		size_t size = pointer->size;
 
-		if (size) {
+		if (size && size <= __MAX_SMALL_SIZE) {
+			__small_free(ptr, size);
+		} else if (size) {
 			cbuf_free(pointer->cbuf_id);
 		}
```

Comment:
If the previous else if is on the same line as the {}, the else should be as well.